database/migrations/2025_01_25_000000_update_player_games_table.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class() extends Migration {
+    public function up(): void
+    {
+        Schema::table('player_games', function (Blueprint $table) {
+            $table->index([
+                'user_id',
+                'achievements_unlocked',
+                'achievements_total',
+                'deleted_at',
+                'game_id',
+            ], 'idx_player_games_suggestions'); // custom name needed because the auto-generated one is too long
+        });
```

Comment:
This looks like the `player_games_completion_sample_idx`, but with the additional `deleted_at` column. Do we need both?

Reply:
There's currently **0** deleted player_games records.

Maybe you could remove the `deleted_at` check form the query and leverage the existing index? A little deleted data isn't going to matter for the outcome of random selection.

Reply:
I believe it's unfortunately still needed:

![Screenshot 2025-01-26 at 10 48 44 AM](https://github.com/user-attachments/assets/a8fc12af-572c-4e55-baad-3a2e9cf06f95)

The screenshot above is with the index removing that column. Eloquent deciding to automagically include the column is what necessitates it being part of the compound index.

Reply:
`->withTrashed()`?

Reply:
That works - fixed in latest.

Reply:
Is the new index still needed?

Reply:
Yes, I think so. As soon as I delete it, I see these monster queries appear:
![Screenshot 2025-01-26 at 10 59 31 AM](https://github.com/user-attachments/assets/ac2cd76a-4f99-45fe-a432-0cc0aef8c75d)

Reply:
If we're not leveraging the old index, there's no reason to remove the `deleted_at` clause.

Reply:
Sorry, I'm happy to implement feedback, but I'm getting confused. What specifically would you like me to do?

My interpretation is you'd like me to add back `deleted_at` to the index as well as the queries, but we don't have any deleted records.

Reply:
I was hoping you'd be able to use the existing `player_games_completion_sample_idx` index, but it sounds like that's not being picked up. 
```
 KEY `player_games_completion_sample_idx` (`game_id`,`achievements_unlocked`,`achievements_total`,`user_id`,`id`),
```
I see now that that index has `id` in it - why would you ever want id in an index? Maybe we can drop it and replace it with yours.

If you're going to have to add a new index, do whatever you feel is necessary with it - like filtering out the deleted items.

Reply:
`player_games_completion_sample_idx` is a [covering index](https://stackoverflow.com/questions/62137/what-is-a-covered-index), but `id` is probably redundant.

Based on what I'm seeing in my local, we don't even need that index anymore and a lot of the code in the strategy can be simplified.

I've updated the new migration and the strategy in latest.